### ISSUE TITLE
Release artifacts (distcheck output) missing for v235

### ISSUE THREAD

**Issue by andreabolognani** [systemd/systemd-bootchart#55]
I'm looking into updating systemd-bootchart to v235 in Fedora and I've noticed that no release tarball (`systemd-bootchart-235.tar.xz`, as produced by `make distcheck`) has been published. This is in contrast with previous releases.

I could of course use the GitHub-generated `systemd-bootchart-235.tar.gz`, and that's in fact what the Debian package seems to be doing already. But that requires introducing a bunch of additional build dependencies (autoconf, automake, intltool) that weren't needed before, so I'm not too keen on doing that.

Is there any chance a proper release tarball could be generated and published?

**Comment by andreabolognani** [comment]
The [Fedora PR](https://src.fedoraproject.org/rpms/systemd-bootchart/pull-request/2) that I've created has been merged in the meantime, so I guess the current arrangement is not considered to be a deal breaker.

I'll leave this open for the time being to give maintainers a chance to explicitly express their intentions when it comes to publishing release tarballs going forward.

Somewhat related: moving from autotools to meson would, in addition to increasing consistency with the main systemd project, make the concept of "release tarball" pretty much completely obsolete. I appreciate that the conversion would require a non-trivial amount of effort.

**Comment by bluca** [comment]
Given XZ and all that, I'm pretty sure we do not want to publish opaque autotools artifacts anymore. All distros should be able to generate them on-the-fly anyway.

**Comment by bluca** [comment]
> Somewhat related: moving from autotools to meson would, in addition to increasing consistency with the main systemd project, make the concept of "release tarball" pretty much completely obsolete. I appreciate that the conversion would require a non-trivial amount of effort.

PRs welcome ;-)

**Comment by andreabolognani** [comment]
> Given XZ and all that, I'm pretty sure we do not want to publish opaque autotools artifacts anymore. All distros should be able to generate them on-the-fly anyway.

@bluca I don't believe the xz situation automatically makes release tarballs undesirable. Most projects that use autotools still publish them AFAIK. They serve a purpose, and the next xz-style attack is probably going to look a lot different anyway.

In any case, the main issue with GitHub-generated archives is that [they are not guaranteed to remain identical over time](https://github.blog/open-source/git/update-on-the-future-stability-of-source-code-archives-and-hashes/), whereas the tooling in various distro builds upon the assumption that they are.

If you really don't want to publish proper release tarballs could you at least generate `git archive` artifacts, possibly sign them, and attach them to the GitHub release? That would ensure that the hash remains consistent over time.

**Comment by bluca** [comment]
Sorry, but no, we are not doing anything manual here - it's the same as the main systemd repo. The git tag is signed, and that has to be enough.
